### PR TITLE
[AIRFLOW-3419] S3_hook.select_key is broken on Python3

### PR DESCRIPTION
**Pull request by vapiravfif**
Fix actual return type of select_key function

Make sure you have checked _all_ steps below.

### Jira

- [X] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3419

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:
Fixes S3_hook.select_key return type to str instead of bytes

### Tests

- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
original test changed to match actual types

### Commits

- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [X] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [X] Passes `flake8`

### CODE REVIEW

File: airflow/hooks/S3_hook.py [hunk 1]
```diff
@@ -272,7 +272,7 @@ def select_key(self, key, bucket_name=None,
             InputSerialization=input_serialization,
             OutputSerialization=output_serialization)
 
-        return ''.join(event['Records']['Payload']
+        return ''.join(event['Records']['Payload'].decode('utf-8')
```
Comment (ashb) [inline airflow/hooks/S3_hook.py]: This possibly needs to be a parameter (with a default value of `utf-8`) just in case not everyone's data is UTF-8

Comment (vapiravfif) [reply airflow/hooks/S3_hook.py]: There are two other hard-coded "utf-8" in S3_hook file, one of them in "read_key" function with same decode.
Changing it to parameter is much bigger scope that this PR

Comment (ashb) [reply airflow/hooks/S3_hook.py]: Ah okay, fair point.

Comment (kurtqq) [reply airflow/hooks/S3_hook.py]: @ashb is that fix will be available in next release?

Comment (ashb) [reply airflow/hooks/S3_hook.py]: Should be now - I've marked it for inclusion in the release

**Comment by Fokko** [comment]
Restarted Travis 👍

**Review by ashb — APPROVED** [review]

**Comment by ashb** [comment]
Sorry I messed up trying to re-base your branch. I didn't mean to close this.

**Comment by ashb** [comment]
Thanks @vapiravfif - I merged this form my own PR - but have kept your authorship on the commit!

**Comment by vapiravfif** [comment]
Thank you, Ash!

**Pull request closed without merging**